File: src/controllers/alertController.js
const db = require("../config/database");

// Hent alle alerts
const getAlerts = async (req, res, next) => {
  try {
    const { device_id, resolved } = req.query;

    let query = "SELECT * FROM alerts WHERE 1=1";
    let params = [];

    if (device_id) {
      query += " AND device_id = ?";
      params.push(device_id);
    }

    if (resolved !== undefined) {
      query += " AND resolved = ?";
      params.push(resolved === "true" ? 1 : 0);
    }

    query += " ORDER BY created_at DESC LIMIT 50";

    const [rows] = await db.query(query, params);
    res.json(rows);
  } catch (err) {
    next(err);
  }
};

// Marker en alert som løst
const resolveAlert = async (req, res, next) => {
  try {
    const { id } = req.params;

    const [result] = await db.query(
      `UPDATE alerts 
       SET resolved = true, resolved_at = NOW() 
       WHERE id = ?`,
      [id]
    );

    if (result.affectedRows === 0) {
      return res.status(404).json({ error: "Alert ikke fundet" });
    }

    res.json({ message: `Alert ${id} er markeret som løst` });
  } catch (err) {
    next(err);
  }
};

module.exports = { getAlerts, resolveAlert };

File: src/controllers/alertController.py
from flask import request, jsonify
from src.config.database import get_db

def get_alerts():
    try:
        device_id = request.args.get("device_id")
        resolved = request.args.get("resolved")

        db = get_db()
        cursor = db.cursor(dictionary=True)

        query = "SELECT * FROM alerts WHERE 1=1"
        params = []

        if device_id:
            query += " AND device_id = %s"
            params.append(device_id)

        if resolved is not None:
            query += " AND resolved = %s"
            params.append(1 if resolved == "true" else 0)

        query += " ORDER BY created_at DESC LIMIT 50"

        cursor.execute(query, params)
        rows = cursor.fetchall()
        cursor.close()
        db.close()

        return jsonify(rows), 200

    except Exception as e:
        return jsonify({"error": str(e)}), 500


def resolve_alert(id):
    try:
        db = get_db()
        cursor = db.cursor()

        cursor.execute(
            "UPDATE alerts SET resolved = true, resolved_at = NOW() WHERE id = %s",
            (id,)
        )
        db.commit()

        if cursor.rowcount == 0:
            return jsonify({"error": "Alert ikke fundet"}), 404

        cursor.close()
        db.close()

        return jsonify({"message": f"Alert {id} er markeret som løst"}), 200

    except Exception as e:
        return jsonify({"error": str(e)}), 500